<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>SimKumar data sets — sce_full_SimKumar4easy • DuoClustering2018</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="SimKumar data sets — sce_full_SimKumar4easy" />
<meta property="og:description" content="Gene counts for scRNA-seq data sets simulated with the splatter package." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DuoClustering2018</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/combine_data_clustering.html">Visualize data sets and clustering results with `iSEE`</a>
    </li>
    <li>
      <a href="../articles/plot_performance.html">Plot performance summaries</a>
    </li>
    <li>
      <a href="../articles/run_clustering.html">Apply clustering method</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>SimKumar data sets</h1>
    
    <div class="hidden name"><code>sce_full_SimKumar.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Gene counts for scRNA-seq data sets simulated with the <code>splatter</code> package.</p>
    </div>

    <pre class="usage"><span class='fu'>sce_full_SimKumar4easy</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_SimKumar4easy</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_SimKumar4easy</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_SimKumar4easy</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_full_SimKumar4hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_SimKumar4hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_SimKumar4hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_SimKumar4hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_full_SimKumar8hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_SimKumar8hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_SimKumar8hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_SimKumar8hard</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>metadata</th>
      <td><p>Logical, whether only metadata should be returned</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Using one subpopulation of the <code>sce_full_Kumar</code> data set as input,
scRNA-seq data with known group structure was simulated with the <code>splatter</code>
package from Zappia et al. (2017). The simulated data have been used to evaluate
the performance of clustering algorithms in Duò et al. (2018).</p>
<p>Three data sets have been generated, each consisting of 500 cells and
approximately 43,000 features, with varying degree of cluster separability.
The <code>sce_full_SimKumar4easy</code> data set consists of 4 subpopulations with
relative abundances 0.1, 0.15, 0.5 and 0.25, and probabilities of differential
expression set to 0.05, 0.1, 0.2 and 0.4 for the four subpopulations,
respectively. The <code>sce_full_SimKumar4hard</code> data set consists of 4
subpopulations with relative abundances 0.2, 0.15, 0.4 and 0.25, and
probabilities of differential expression 0.01, 0.05, 0.05 and 0.08. Finally,
the <code>sce_full_SimKumar8hard</code> data set consists of 8 subpopulations with
relative abundances 0.13, 0.07, 0.1, 0.05, 0.4, 0.1, 0.1 and 0.05, and
probabilities of differential expression equal to 0.03, 0.03, 0.03, 0.05, 0.05,
0.07, 0.08 and 0.1, respectively.</p>
<p>The <code>scater</code> package was used to perform quality control of the data sets
(McCarthy et al. (2017)).
Features with zero counts across all cells, as well as cells with total count
or total number of detected features more than 3 median absolute deviations
(MADs) below the median across all cells (on the log scale), were excluded.
The <code>filteredExpr</code>, <code>filteredHVG</code> and <code>filteredM3Drop10</code> are
further reduced data sets. For each of the filtering method, we retained 10
percent of the original number of genes
(with a non-zero count in at least one cell) in the original data sets.</p>
<p>For the <code>filteredExpr</code> data sets, only the genes with the highest average
expression (log-normalized count) value across all cells were retained.
Using the <code>Seurat</code> package, the <code>filteredHVG</code> data sets were filtered
on the variability of the features and only the most highly variable ones were
retained (Satija et al. (2015)).  Finally, the <code>M3Drop</code> package was used
to model the dropout rate of the genes as a function of the mean expression
level using the Michaelis-Menten equation and select variables to retain for
the <code>filteredM3Drop10</code> data sets (Andrews and Hemberg (2018)).</p>
<p>The <code>scater</code> package was used to normalize the count values, based on
normalization factors calculated by the deconvolution method from the
<code>scran</code> package (Lun et al. (2016)).
This data set is provided as a <code>SingleCellExperiment</code> object
(Lun and Risso (2017)). For further information on the
<code>SingleCellExperiment</code> class, see the corresponding manual.</p>
    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>SingleCellExperiment</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a <code>SingleCellExperiment</code> object.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Andrews, T.S., and Hemberg, M. (2018). <em>Dropout-based feature selection
for scRNASeq</em>. bioRxiv doi:https://doi.org/10.1101/065094.</p>
<p>Duò, A., Robinson, M.D., and Soneson, C. (2018).
<em>A systematic performance evaluation of clustering methods for single-cell
RNA-seq data.</em>
F1000Res. 7:1141.</p>
<p>Lun, A.T.L., Bach, K., and Marioni, J.C. (2016) <em>Pooling across cells to
normalize single-cell RNA sequencing data with many zero counts.</em>
Genome Biol. 17(1): 75.</p>
<p>Lun, A.T.L., and Risso, D. (2017). <em>SingleCellExperiment: S4 Classes for
Single Cell Data</em>. R package version 1.0.0.</p>
<p>McCarthy, D.J., Campbell, K.R., Lun, A.T.L., and Wills, Q.F. (2017):
<em>Scater: pre-processing, quality control, normalization and visualization
of single-cell RNA-seq data in R.</em> Bioinformatics 33(8): 1179-1186.</p>
<p>Satija, R., Farrell, J.A., Gennert, D., Schier, A.F., and Regev, A. (2015).
<em>Spatial reconstruction of single-cell gene expression data.</em>
Nat. Biotechnol. 33(5): 495–502.</p>
<p>Zappia, L., Phipson, B., and Oshlack, A. (2017). <em>Splatter: simulation of
single-cell RNA sequencing data.</em> Genome Biol. 18(1): 174.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>sce_filteredExpr10_SimKumar4easy</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2020-10-02</span></div><div class='output co'>#&gt; <span class='message'>see ?DuoClustering2018 and browseVignettes('DuoClustering2018') for documentation</span></div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; class: SingleCellExperiment 
#&gt; dim: 4360 500 
#&gt; metadata(2): params log.exprs.offset
#&gt; assays(8): BatchCellMeans BaseCellMeans ... logcounts normcounts
#&gt; rownames(4360): Gene30490 Gene22336 ... Gene10402 Gene18379
#&gt; rowData names(16): Gene BaseGeneMean ... total_counts
#&gt;   log10_total_counts
#&gt; colnames(500): Cell1 Cell2 ... Cell499 Cell500
#&gt; colData names(16): Cell Batch ... pct_counts_top_500_features
#&gt;   is_cell_control</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Angelo Duò.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


