<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Trapnell data sets — sce_full_Trapnell • DuoClustering2018</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Trapnell data sets — sce_full_Trapnell" />
<meta property="og:description" content="Gene or TCC counts for scRNA-seq data set from Trapnell et al. (2014),
consisting of primary myoblasts over a time course of serum-induced
differentiation." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DuoClustering2018</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/combine_data_clustering.html">Visualize data sets and clustering results with `iSEE`</a>
    </li>
    <li>
      <a href="../articles/plot_performance.html">Plot performance summaries</a>
    </li>
    <li>
      <a href="../articles/run_clustering.html">Apply clustering method</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Trapnell data sets</h1>
    
    <div class="hidden name"><code>sce_full_Trapnell.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Gene or TCC counts for scRNA-seq data set from Trapnell et al. (2014),
consisting of primary myoblasts over a time course of serum-induced
differentiation.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>sce_full_Trapnell</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_Trapnell</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_Trapnell</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_Trapnell</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_full_TrapnellTCC</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_TrapnellTCC</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_TrapnellTCC</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_TrapnellTCC</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>metadata</th>
      <td><p>Logical, whether only metadata should be returned</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This is a scRNA-seq data set originally from Trapnell et al. (2014). The data
set consists of gene-level read counts or TCCs (transcript compatibility counts)
from human primary myoblasts over a time course of serum-induced
differentiation. It contains 3 subpopulations, defined by the cell phenotype
given by the authors' annotations. The data sets have been used to evaluate the
performance of clustering algorithms in Duò et al. (2018).</p>
<p>For the <code>sce_full_Trapnell</code> data set, all genes except those with zero
counts across all cells are retained. The gene counts are
gene-level length-scaled TPM values derived from Salmon (Patro et al. (2017))
quantifications (see
Soneson and Robinson (2018)). For the TCC data set we estimated transcripts
compatibility counts using <code>kallisto</code> as an alternative to the gene-level
count matrix (Bray et al. (2016), Ntranos et al. (2016)).</p>
<p>The <code>scater</code> package was used to perform quality control of the data sets
(McCarthy et al. (2017)).
Features with zero counts across all cells, as well as all cells with total
count or total number of detected features more than 3 median absolute
deviations (MADs) below the median across all cells (on the log scale),
were excluded. Additionally, cells that were classified as doublets or
debris were filtered out.</p>
<p>The <code>sce_full_Trapnell</code> data set consists of 222 cells and 41,111 features,
the <code>sce_full_TrapnellTCC</code> data set of 227 cells and 684,953 features,
respectively.
The <code>filteredExpr</code>, <code>filteredHVG</code> and <code>filteredM3Drop10</code> are
further reduced data sets.
For each of the filtering method, we retained 10 percent of the original
number of genes
(with a non-zero count in at least one cell) in the original data sets.</p>
<p>For the <code>filteredExpr</code> data sets, only the genes/TCCs with the highest
average expression (log-normalized count) value across all cells were retained.
Using the <code>Seurat</code> package, the <code>filteredHVG</code> data sets were filtered
on the variability of the features and only the most highly variable ones were
retained (Satija et al. (2015)). Finally, the <code>M3Drop</code> package was used
to model the dropout rate of the genes as a function of the mean expression
level using the Michaelis-Menten equation and select variables to retain for
the <code>filteredM3Drop10</code> data sets (Andrews and Hemberg (2018)).</p>
<p>The <code>scater</code> package was used to normalize the count values, based on
normalization factors calculated by the deconvolution method from the
<code>scran</code> package (Lun et al. (2016)).</p>
<p>This data set is provided as a <code>SingleCellExperiment</code> object
(Lun and Risso (2017)). For further information on the
<code>SingleCellExperiment</code> class, see the corresponding manual.
Raw data files for the original data set (GSE52529) are available from
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52529.</p>
    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>SingleCellExperiment</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a <code>SingleCellExperiment</code> object.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Andrews, T.S., and Hemberg, M. (2018). <em>Dropout-based feature selection
for scRNASeq</em>. bioRxiv doi:https://doi.org/10.1101/065094.</p>
<p>Bray, N.L., Pimentel, H., Melsted, P., and Pachter, L. (2016).
<em>Near-optimal probabilistic RNA-seq quantification</em>.
Nat. Biotechnol. 34: 525–527.</p>
<p>Duò, A., Robinson, M.D., and Soneson, C. (2018).
<em>A systematic performance evaluation of clustering methods for single-cell
RNA-seq data.</em>
F1000Res. 7:1141.</p>
<p>Lun, A.T.L., Bach, K., and Marioni, J.C. (2016) <em>Pooling across cells to
normalize single-cell RNA sequencing data with many zero counts.</em>
Genome Biol. 17(1): 75.</p>
<p>Lun, A.T.L., and Risso, D. (2017). <em>SingleCellExperiment: S4 Classes for
Single Cell Data</em>. R package version 1.0.0.</p>
<p>McCarthy, D.J., Campbell, K.R., Lun, A.T.L., and Wills, Q.F. (2017):
<em>Scater: pre-processing, quality control, normalization and visualization
of single-cell RNA-seq data in R.</em> Bioinformatics 33(8): 1179-1186.</p>
<p>Ntranos, V., Kamath, G.M., Zhang, J.M., Pachter, L., and Tse, D.N. (2016):
<em>Fast and accurate single-cell RNA-seq analysis by clustering of
transcript-compatibility counts.</em> Genome Biol. 17:112.</p>
<p>Patro, R., Duggal, G., Love, M.I., Irizarry, R.A., and Kingsford, C. (2017):
<em>Salmon provides fast and bias-aware quantification of transcript
expression.</em> Nat. Methods 14:417-419.</p>
<p>Satija, R., Farrell, J.A., Gennert, D., Schier, A.F., and Regev, A. (2015).
<em>Spatial reconstruction of single-cell gene expression data.</em>
Nat. Biotechnol. 33(5): 495–502.</p>
<p>Soneson, C., and Robinson, M.D. (2018). <em>Bias, robustness and scalability
in single-cell differential expression analysis.</em> Nat. Methods, 15(4): 255-261.</p>
<p>Trapnell, C., Cacchiarelli, D., Grimsby, J., Pokharel, P., Li, S., Morse, M.,
Lennon, N.J., Livak, K.J., Mikkelsen, T.S., and Rinn, J.L. (2014).
<em>The dynamics and regulators of cell fate decisions are revealed by
pseudotemporal ordering of single cells.</em> Nat. Biotechnol. 32(4): 381–386.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='fu'>sce_filteredExpr10_Trapnell</span><span class='op'>(</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> see ?DuoClustering2018 and browseVignettes('DuoClustering2018') for documentation</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: SingleCellExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 4111 222 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(1): log.exprs.offset</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(3): counts logcounts normcounts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames(4111): ENSG00000198804.2 ENSG00000210082.2 ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   ENSG00000185404.16 ENSG00000168393.12</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(8): is_feature_control mean_counts ... total_counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   log10_total_counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(222): GSM1268960 GSM1268961 ... GSM1269246 GSM1269247</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(61): title geo_accession ... pct_counts_top_500_features</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   is_cell_control</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> reducedDimNames(2): PCA TSNE</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> mainExpName: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> altExpNames(0):</span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Angelo DuÃ².</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


