<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Zheng data sets — sce_full_Zhengmix4eq • DuoClustering2018</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Zheng data sets — sce_full_Zhengmix4eq" />
<meta property="og:description" content="Gene counts for scRNA-seq data sets from Zheng et al. (2017), consisting of
pre-sorted cell types combined into three artificial data sets with different
cell proportions." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DuoClustering2018</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/combine_data_clustering.html">Visualize data sets and clustering results with `iSEE`</a>
    </li>
    <li>
      <a href="../articles/plot_performance.html">Plot performance summaries</a>
    </li>
    <li>
      <a href="../articles/run_clustering.html">Apply clustering method</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Zheng data sets</h1>
    
    <div class="hidden name"><code>sce_full_Zhengmix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Gene counts for scRNA-seq data sets from Zheng et al. (2017), consisting of
pre-sorted cell types combined into three artificial data sets with different
cell proportions.</p>
    </div>

    <pre class="usage"><span class='fu'>sce_full_Zhengmix4eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_Zhengmix4eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_Zhengmix4eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_Zhengmix4eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_full_Zhengmix4uneq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_Zhengmix4uneq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_Zhengmix4uneq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_Zhengmix4uneq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_full_Zhengmix8eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredExpr10_Zhengmix8eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredHVG10_Zhengmix8eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'>sce_filteredM3Drop10_Zhengmix8eq</span><span class='op'>(</span>metadata <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>metadata</th>
      <td><p>Logical, whether only metadata should be returned</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This is a scRNA-seq data set originally from Zheng et al. (2017). The data set
consists of eight pre-sorted cell types (B-cells, naive cytotoxic T-cells, CD14
monocytes, regulatory T-cells, CD56 NK cells, memory T-cells, CD4 T-helper cells
and naive T-cells) from <em>Homo sapiens</em> combined into three artificial data
sets with different cell proportions. The annotated cell type (obtained by
pre-sorting of the cells) is used as the true cell label. The data sets have
been used to evaluate the performance of clustering algorithms in Duò et al.
(2018).</p>
<p>For the <code>Zhengmix4eq</code> data set, randomly selected B-cells, CD14 monocytes,
naive cytotoxic T-cells and regulatory T-cells were combined in equal
proportions (1,000 cells per subpopulation).
The <code>Zhengmix4uneq</code> data set consists of four cell types, combined in
unequal proportions (1,000 B-cells, 500 naive cytotoxic T-cells, 2,000 CD14
monocytes and 3,000 regulatory T-cells). For the <code>Zhengmix8eq</code> data set,
all eight populations were combined in approximately equal proportions (400–600
cells per population).</p>
<p>For the <code>sce_full_Zhengmix4eq</code>, <code>sce_full_Zhengmix4uneq</code> and
<code>sce_full_Zhengmix8eq</code> data set, all genes except those with zero counts
across all cells are retained. The gene counts are unique molecular
identifiers (UMIs) counts.
The <code>scater</code> package was used to perform quality control of the data
(McCarthy et al. (2017)).
Features with zero counts across all cells, as well as all cells with total
count or total number of detected features more than 3 median absolute
deviations (MADs) below the median across all cells (on the log scale),
were excluded.</p>
<p>The <code>sce_full_Zhengmix4eq</code> data set consists of 3,994 cells and 15,568
features, the <code>sce_full_Zhengmix4uneq</code> data set of 6,498 cells and 16,443
features and the <code>sce_full_Zhengmix8eq</code> of 3,994 cells and 16,443 features,
respectively.
The <code>filteredExpr</code>, <code>filteredHVG</code> and <code>filteredM3Drop10</code> are
further reduced data sets.
For each of the filtering method, we retained 10 percent of the original
number of genes
(with a non-zero count in at least one cell) in the original data sets.</p>
<p>For the <code>filteredExpr</code> data sets, only the genes with the highest average
expression (log-normalized count) value across all cells were retained.
Using the <code>Seurat</code> package, the <code>filteredHVG</code> data sets were filtered
on the variability of the features and only the most highly variable ones were
retained (Satija et al. (2015)).  Finally, the <code>M3Drop</code> package was used
to model the dropout rate of the genes as a function of the mean expression
level using the Michaelis-Menten equation and select variables to retain for the
<code>filteredM3Drop10</code> data sets (Andrews and Hemberg (2018)).</p>
<p>The <code>scater</code> package was used to normalize the count values, based on
normalization factors calculated by the deconvolution method from the
<code>scran</code> package (Lun et al. (2016)).
This data set is provided as a <code>SingleCellExperiment</code> object
(Lun and Risso (2017)). For further information on the
<code>SingleCellExperiment</code> class, see the corresponding manual.
Raw data files or the original data sets are available from
https://support.10xgenomics.com/single-cell-gene-expression/datasets.</p>
    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>SingleCellExperiment</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a <code>SingleCellExperiment</code> object.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Andrews, T.S., and Hemberg, M. (2018). <em>Dropout-based feature selection for
scRNASeq</em>. bioRxiv doi:https://doi.org/10.1101/065094.</p>
<p>Duò, A., Robinson, M.D., and Soneson, C. (2018).
<em>A systematic performance evaluation of clustering methods for single-cell
RNA-seq data.</em>
F1000Res. 7:1141.</p>
<p>Lun, A.T.L., Bach, K., and Marioni, J.C. (2016) <em>Pooling across cells to
normalize single-cell RNA sequencing data with many zero counts.</em>
Genome Biol. 17(1): 75.</p>
<p>Lun, A.T.L., and Risso, D. (2017). <em>SingleCellExperiment: S4 Classes for
Single Cell Data</em>. R package version 1.0.0.</p>
<p>McCarthy, D.J., Campbell, K.R., Lun, A.T.L., and Wills, Q.F. (2017):
<em>Scater: pre-processing, quality control, normalization and visualization
of single-cell RNA-seq data in R.</em> Bioinformatics 33(8): 1179-1186.</p>
<p>Satija, R., Farrell, J.A., Gennert, D., Schier, A.F., and Regev, A. (2015).
<em>Spatial reconstruction of single-cell gene expression data.</em>
Nat. Biotechnol. 33(5): 495–502.</p>
<p>Zheng, G.X., Terry, J.M., Belgrader P., Ryvkin, Pl, Bent, Z.W., Wilson, R.,
Ziraldo, S.B., Wheeler, T.D., McDermott, G.P., Zhu, J., Gregory, M.T.,
Shuga, J., Montesclaros, L., Underwood, J.G., Masquelier, D.A., Nishimura, S.Y.,
Schnall-Levin, M., Wyatt, P.W., Hindson, C.M., Bharadwaj, R., Wong, A.,
Ness, K.D., Beppu, L.W., Deeg, H.J., McFarland, C., Loeb, K.R., Valente, W.J.,
Ericson, N.G., Stevens, E.A., Radich, J.P., Mikkelsen, T.S., Hindson, B.J.,
and Bielas, J.H. (2017). <em>Massively parallel
digital transcriptional profiling of single cells.</em> Nat. Commun. 8:14049.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>sce_filteredExpr10_Zhengmix4eq</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2021-03-24</span></div><div class='output co'>#&gt; <span class='message'>see ?DuoClustering2018 and browseVignettes('DuoClustering2018') for documentation</span></div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; class: SingleCellExperiment 
#&gt; dim: 1556 3555 
#&gt; metadata(1): log.exprs.offset
#&gt; assays(3): counts logcounts normcounts
#&gt; rownames(1556): ENSG00000167526 ENSG00000140988 ... ENSG00000011009
#&gt;   ENSG00000168522
#&gt; rowData names(10): id symbol ... total_counts log10_total_counts
#&gt; colnames(3555): b.cells6276 b.cells6144 ... regulatory.t1084
#&gt;   regulatory.t9696
#&gt; colData names(14): dataset barcode ... pct_counts_top_500_features
#&gt;   is_cell_control
#&gt; reducedDimNames(2): PCA TSNE
#&gt; mainExpName: NULL
#&gt; altExpNames(0):</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Angelo Duò.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


